gprofiler/profilers/perf.py

```diff
@@ -276,6 +289,11 @@ def snapshot(self) -> ProcessToProfileData:
             ).items()
         }
 
+        for perf in self._perfs:
```

Comment:
It's always good to guard over features (especially very "active" features like this one) with a feature flag. Let's add one here? It can be enabled by default, and we add a flag `--perf-no-restart` that disables it.